source/blog/_posts/2018-08-23-against-microservices.md

```diff
@@ -0,0 +1,136 @@
+
+And there are real world benefits to all of them! Good old static code analysis will help you greatly with dependency management and finally this can all be part of the global CI process.
+
+And it even might be necessary to create different Applications for parts of your problem. Mayber the requirements for a single part of your application are so drastically different that this becomes necessary. Feel free to use concepts from Microservices. **Hybridization** of architecture is not necessarily a bad thing.  
+
+Problem solved? Well there are still the organizational difficulties...
+
+## Alternative: (Micro-)Kernel
+
+Microservices and monoliths are on oposing sides of the static dependency spectrum. While a monolith is interconnected and interdependent a Microservice is *(almost)* not. Well this spectrum does not only consist of black and white. My favorite alternative is the kernel style. A Kernel represents a core domain of an application. In our case eCommerce. This kernel then provides an extension mechanism for others to extend, alter and replace these core concepts. A plugin system if you want.
+
+Apart from the basic eCommerce workflows the Kernel provides the technical groundwork for all plugins to use. It contains a deployment machanism, package management and lays the technological groundwork for the necessarry infrastructure coupling. Most importantly it provides **the base quality of the product as a whole**.  
+
+In a company driven by Conways law the natural limit for a kernel is the amount of features a single team can handle. A basic runtime that shares a technical and functional base for everyone to harness. One arranges orthogonal features in orthogonal teams and creates a core that is owned and used by everyone. The viability of this approach is proven by big and small companies alike, think Operating System vendors, Framework vendors and the like. There is however a critical situation when a kernel's size outgrows the team. A need for action arises. A split is necessary.
+
+Conway's insight should mean that an organization in order to produce software needs to be fluid enough to support good design. So if an application changes the communication and therefore team structure needs to be changed in the best interest of the product.      
+
+## Conclusion
+
+From a technical standpoint there is no clear benefit in using Microservices as the main application design. They just have the potential to add risk and cost to a project. In my opinion this style just complicates stuff from the get go. Sizing is a really hard problem in software and I have seen many applications that where suprisingly large in relation to the work they actually performed. If you have a fairly simple problem to solve a single developer will be faster to fix it them twenty. Or as we say: 
```

Comment:
I think there is a comma missing: If you have a fairly simple problem to solve, a single developer will be faster to fix it them twenty.